hlf/scala/general-communication.md

```diff
@@ -65,35 +65,56 @@ The ADMIN can then
-The proposalBytes have to be relayed to the USER to sign it.
+The proposalBytes have to be relayed to the USER to sign it. The proposal itself is for the transaction defined in [ApproveOperation](./operation.md#ApproveOperation); an approval for the initiated transaction, to be signed by the user.
 **_IMPORTANT: When asking for a proposal, the credentials used for setting up the initial connection are used to approve the transaction as well (in this case as the ADMIN)_**
 
-- pass the proposalBytes on to the USER who wants to sign it
-- have the USER create his signature for the "unsigned" proposalBytes
+- pass the proposalBytes with approvalResult on to the USER who wants to sign it
+- have the USER create their signature for the "unsigned" proposalBytes
 - receive the signature from the USER (together with the original proposalBytes)
 
-- submit the signed proposal ("unsigned" proposalBytes, signatureBytes) to retrieve an usigned Transaction
+- submit the signed proposal ("unsigned" proposalBytes, signatureBytes) to retrieve an usigned Transaction.
+> While the proposal is fetched from the specific connections (e.g. MatriculationConnection, CertificateConnection, ...), the getUnsignedTransaction, which submits the signed proposal, is called on the OperationConnection.
 ```scala
-val transactionBytes = certificateConnection.getUnsignedTransaction(proposalBytes: Array[Byte], signatureBytes: Array[Byte])
+val transactionBytes = operationConnection.getUnsignedTransaction(proposalBytes: Array[Byte], signatureBytes: Array[Byte])
 ```
 
 - pass the transactionBytes on to the USER who wants to sign it
 - have the USER create his signature for the "unsigned" transactionBytes
 - receive the signature from the USER (together with the original transactionBytes)
 
 - submit the signed transaction ("unsigned" transactionBytes, signatureBytes)
+>The Connection receiving the signedTransaction is always the OperationConnection.
 ```scala
-val (approvalResult, realTransactionResult) = certificateConnection.submitSignedTransaction(transactionBytes: Array[Byte], signatureBytes: Array[Byte])
+val approvalResult = operationConnection.submitSignedTransaction(transactionBytes: Array[Byte], signatureBytes: Array[Byte])
+
+val transactionResult = operationConnection.executeTransaction(jsonOperationData: String, timeoutMilliseconds: int = 5000)
 ```
-The approvalResult is a JsonObject [OperationData](../chaincode/operation.md#OperationData).
-The realTransactionResult is the chaincode response to the desired transaction
-(e.g. an InsufficientApproval Error, or information about the successfully changed ledger state)
+The approvalResult is a JsonObject [OperationData](../chaincode/operation.md#OperationData). 
+Invoking executeTransactionFromOperation executes the transaction described by the transactionInfo in the OperationData.
+In case of a NetworkException, the transaction is retried until the specified timeout is reached.
```

Comment:
on implementation side, how is this implemented? Do you just retry the whole time? (the call might terminate in microseconds if the socket is not available, so this is really really bad busy waiting, is there a timeout between tries?)

Reply:
this is not implemented at all, and we have not even considered pros and cons of different approaches.
Regarding a timeout after X attempts - I was specifically told to not do that